src/App.tsx

```diff
@@ -27,33 +31,193 @@ export const GET_POSTS = gql`
+        <h3 className="w-[80%] self-center text-[20px]">
+          Create Post
+        </h3>
+        <label className="w-[80%] self-center">
+          Title
+        </label>
+        <input placeholder="title..." className="w-[80%] self-center border-2 p-2 rounded-md" value={newPost.title} onChange={(event) => newPostTitleChange(event.target.value)} />
+        <label className="w-[80%] self-center">
+          Content
+        </label>
+        <textarea cols={20} placeholder="Content..." className="w-[80%] self-center border-2 p-2 rounded-md" value={newPost.body} onChange={(event) => newPostBodyChange(event.target.value)} />
+        <div className="absolute right-4 bottom-4 flex flex-row w-[20%] justify-between">
+          <button className="rounded border-2  p-2 " onClick={() => props.close()} > cancel </button>
+          <button className="rounded border-2  p-2 bg-[#0b57d0] text-white" onClick={() => props.submit(newPost)} > Send </button>
+        </div>
+      </div>
+    </div>
+  )
+}
 
-const NewPostItem = (post: Post) => (
-  <div key={post.id} className="bg-secondary rounded-md px-2 py-1 mt-4">
-    <h2>{post.title}</h2>
-    <p className="text-dark-body text-sm m-0">{post.body}</p>
+const NewPostItem = (props: { post: PostProps, key: number, onClick: () => void }) => (
+  <div key={props.post.id} className="bg-secondary rounded-md px-2 py-1 mt-4">
+    <h2>{props.post.title}</h2>
+    <p className="text-dark-body text-sm m-0">{props.post.body}</p>
   </div>
 );
 
 export const Main = () => {
-  const { loading, error, data } = useQuery(GET_POSTS);
+
+  //State used to  store posts and apply changes in front-end
+  const [posts, setPosts] = useState<PostProps[]>([]);
+
+  //State used to verify if we are updating a post
+  const [isUpdatingPost, setIsUpdatingPost] = useState(false);
+  //State used to display the create or update Post modal
+  const [showUpdateCreateModal, setShowUpdateCreateModal] = useState(false);
+
+  //Hook to use createPost mutation
+  const [createPost] = useMutation(CREATE_POST);
+
+  //Hook to use updatePost mutation
+  const [updatePost] = useMutation(UPDATE_POST);
+
+  //Hook to use deletePost mutation
+  const [deletePost] = useMutation(DELETE_POST);
+
+  //Hook to use Get posts Query
+  const { loading, error, data } = useQuery(GET_POSTS, {
+
+    //Variables for get posts query like the limit per page and the sort field and order
+    variables: { options: { slice: { limit: 10 }, sort: { field: "id", order: "DESC" } } },
+
+    //The function that executes once the query is successfull
+    onCompleted: () => {
+      setPosts(data?.posts?.data.map((post: Post) => {
+        return { ...post, isSelected: false, isNew: false }
+      }))
+    }
+  });
+
+  //The function that executes from the modal to send to GQL the new Data for update or create
+  const createOrUpdatePost = (newPost: CreatePostProps) => {
```

Comment:
This is an interesting solution. How would you go about testing for this function?